### ISSUE TITLE
Feature: Warn when function has been defined more than once

### ISSUE THREAD

**Issue by Jonta** [koalaman/shellcheck#2412]
#### For new checks and feature suggestions
- [x] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/sh

print_greeting () {
        echo "Hi"
}

print_greeting () {
        echo "Hello"
}

print_greeting

```

#### Here's what shellcheck currently says:

"No issues detected"

#### Here's what I wanted or expected to see:

"print_greeting has been defined twice. Perhaps unintentionally"

**Comment by guillaume-d** [comment]
Looks like a duplicate of #2316.
